akka-contrib/src/main/scala/akka/contrib/circuitbreaker/CircuitBreakerActor.scala

```diff
@@ -0,0 +1,247 @@
+   * [[akka.contrib.circuitbreaker.CircuitBreakerActor.TellOnly]]
+   *
+   * The circuit breaker implements the same state machine documented in [[akka.pattern.CircuitBreaker]]
+   *
+   * @param target the actor to proxy
+   * @param maxFailures maximum number of failures before opening the circuit
+   * @param callTimeout timeout before considering the ongoing call a failure
+   * @param resetTimeout time after which the channel will be closed after entering the open state
+   * @param circuitEventListener an actor that will receive a series of messages of type
+   *                             [[akka.contrib.circuitbreaker.CircuitBreakerActor.CircuitBreakerEvent]]
+   * @param failureDetector  function to detect if the a message received from the target actor as
+   *                         response from a request represent a failure
+   * @param failureMap   function to map a failure into a response message. The failing response message is wrapped
+   *                     into a [[akka.contrib.circuitbreaker.CircuitBreakerActor.CircuitOpenFailure]] object
+   */
+  def props(target: ActorRef,
+            maxFailures: Int,
+            callTimeout: Timeout,
+            resetTimeout: Timeout,
+            circuitEventListener: Option[ActorRef],
+            failureDetector: Any ⇒ Boolean,
+            failureMap: CircuitOpenFailure ⇒ Any) =
+    Props(new CircuitBreakerActor(target, maxFailures, callTimeout, resetTimeout, circuitEventListener, failureDetector, failureMap))
+
+  sealed trait CircuitBreakerCommand
+
+  final case class TellOnly(msg: Any) extends CircuitBreakerCommand
+
+  sealed trait CircuitBreakerResponse
+  final case class CircuitOpenFailure(failedMsg: Any)
+
+  sealed trait CircuitBreakerEvent
+  final case class CircuitOpen(circuit: ActorRef) extends CircuitBreakerCommand
+  final case class CircuitClosed(circuit: ActorRef) extends CircuitBreakerCommand
+  final case class CircuitHalfOpen(circuit: ActorRef) extends CircuitBreakerCommand
+
+  sealed trait CircuitBreakerState
+  case object Open extends CircuitBreakerState
+  case object Closed extends CircuitBreakerState
+  case object HalfOpen extends CircuitBreakerState
+
+  final case class CircuitBreakerStateData(failureCount: Int = 0, firstHalfOpenMessageSent: Boolean = false)
+
+  final case class CircuitBreakerActorBuilder(
+    maxFailures: Int, callTimeout: Timeout, resetTimeout: Timeout,
+    circuitEventListener: Option[ActorRef] = None,
+    failureDetector: Any ⇒ Boolean = { _ ⇒ false },
+    openCircuitFailureConverter: CircuitOpenFailure ⇒ Any = identity) {
+
+    /**
+     * Creates the props for a [[akka.contrib.circuitbreaker.CircuitBreakerActor]] proxying the given target
+     *
+     * @param target the target actor ref
+     */
+    def propsForTarget(target: ActorRef) = CircuitBreakerActor.props(target, maxFailures, callTimeout, resetTimeout, circuitEventListener, failureDetector, openCircuitFailureConverter)
```

Comment:
just `props()` should be enough I think (one can call it like `props(target = ref)` if one wants to)